composer/datasets/streaming/writer.py

```diff
@@ -82,29 +88,63 @@ def __init__(self, dirname: str, fields: List[str], shard_size_limit: int = 1 <<
         self.new_samples = []
         self.new_shard_size = 0
 
+        # compression scheme for shards
+        self.compression_scheme = None
+        self.open_f = lambda n: open(n, 'xb')
+        self._set_compression_settings(compression)
```

Comment:
I like that this parsing is done via a helper. But generally speaking, it'd be preferable to have a function do one thing and only one thing. Could this work?

```suggestion
        self.compression_scheme = _get_compression_scheme(compression)
```

Then, `_get_compression_scheme` could be a static helper function defined at the top of the file (need not be an instance method, but ok with me if it is)